CONTRIBUTING.md

```diff
@@ -94,137 +65,217 @@ script before each commit and abort the commit on error. To
 
-- Navigate to the '/RetrospectiveExtension.Frontend' folder, run `npm install` to download all the dependent packages listed in 'package.json'.
+---
 
-- Run `npm run build:d` to build the project.
+- The first time this repository is opened in Visual Studio Code, the .devcontainer folder will be detected. In the bottom right-hand corner of the screen, a prompt will be displayed: “Folder contains a Dev Container configuration file. Reopen folder to develop in a container.”.
+- Some commands, such as `dotnet build` and `npm run build:d` may need `sudo` prefixed to execute successfully.
 
-- Run `npm run start:dev` to start the webpack-dev-server
+### Github Codespaces
 
-- Start debugger (making sure the webpack-dev-server is still running). The default launch configuration should be set to Launch Firefox.
+1. If necessary, read Github's [codespace creation](https://docs.github.com/en/codespaces/developing-in-codespaces/creating-a-codespace) documentation.
 
-- Once Firefox starts up, you should get an untrusted certificate error page. Select Advanced and then select **Accept the Risk and Continue** and log into your Azure DevOps account. From now on, if you leave this Firefox window open, the debugger will reattach instead of starting a clean Firefox instance each time.
+2. Create a branch for the work you are planning to complete.
 
-- Once you are logged in to Azure DevOps, your extension should be running. Set a breakpoint in a method in VS Code and you should see that breakpoint hit when that method executes.
+3. Create a codespace through the new branch's dropdown on Github.
 
-### Storage
+4. Codespaces are powered by [Visual Studio Code and dev containers](https://docs.github.com/en/codespaces/developing-in-codespaces/developing-in-a-codespace) - once the codespace has been entered, you can follow the instructions within the [dev containers section](#dev-containers) for continuing to set up the development environment. View the [Developing in a codespace](https://docs.github.com/en/codespaces/developing-in-codespaces/developing-in-a-codespace) documentation for additional details.
 
-The Retrospectives tool uses the [Azure DevOps data service](https://docs.microsoft.com/en-us/azure/devops/extend/develop/data-storage?view=vsts) for handling all its storage.
+## Build
+
+### Test in Azure DevOps
+
+Test changes in the Azure DevOps environment by publishing a development version of the extension under an Azure DevOps publisher account.
+
+---
+
+1. Clone this repository, and open in your preferred [development environment](#development-environments).
+2. Using Powershell, navigate to the '/RetrospectiveExtension.Frontend' folder, run `npm install`. This will download all the dependent packages listed in 'package.json'.
+3. Copy the file `RetrospectiveExtension.Frontend\config\environment.tsx.template` into `RetrospectiveExtension.Frontend\config\environment.tsx` and update the fields
```

Comment:
Just a note: this part is subject to change depending on Mohamed's PR.

Reply:
Yes - I figured I would merge my changes after his - since it will be easier for me to figure out where the changes since I updated this doc